app/src/main/java/com/codepath/apps/critterfinder/network/PetFinderHttpClient.java

```diff
@@ -29,14 +29,25 @@ public PetFinderHttpClient() {
     }
 
     // PetFinder get list of pets
-    public void findPetList(JsonHttpResponseHandler handler) {
+    public void findPetList(JsonHttpResponseHandler handler,SearchFilter searchFilter) {
         String apiUrl = getApiUrl("pet.find");
         // Can specify query string params directly or through RequestParams.
         RequestParams params = new RequestParams();
         params.put("format", "json");
-//		params.put("count", 25);
+		params.put("count", 25);
         params.put("key", REST_CONSUMER_KEY);
-        params.put("location", "94941");        // TO DO don't hard code Zip Code
+        if (searchFilter.getPostalCode() != null)
+            params.put("location", searchFilter.getPostalCode());
+        else    // location is a REQUIRED field
+            params.put("location","94140");
+        if (searchFilter.getSpecies() != null)
+            params.put("animal", searchFilter.getSpecies());
```

Comment:
oh interesting! i thought we'd have to convert the enum to a string value - like getSpecies.getName() to make this work. That's cool.